prow/plugins/plugins.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugins
+
+import (
+	"fmt"
+	"io/ioutil"
+	"sync"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/ghodss/yaml"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/jobs"
+	"k8s.io/test-infra/prow/kube"
+)
+
+var (
+	allPlugins           = map[string]struct{}{}
+	issueCommentHandlers = map[string]IssueCommentHandler{}
+	pullRequestHandlers  = map[string]PullRequestHandler{}
+)
+
+type IssueCommentHandler func(*PluginAgent, github.IssueCommentEvent) error
+
+func RegisterIssueCommentHandler(name string, fn IssueCommentHandler) {
+	allPlugins[name] = struct{}{}
+	issueCommentHandlers[name] = fn
+}
+
+type PullRequestHandler func(*PluginAgent, github.PullRequestEvent) error
+
+func RegisterPullRequestHandler(name string, fn PullRequestHandler) {
+	allPlugins[name] = struct{}{}
+	pullRequestHandlers[name] = fn
+}
+
+type PluginAgent struct {
+	GitHubClient *github.Client
+	KubeClient   *kube.Client
+	JobAgent     *jobs.JobAgent
+
+	mut sync.Mutex
+	// Repo (eg "k/k") -> list of handler names.
+	ps map[string][]string
+}
+
+// Load attempts to load config from the path. It returns an error if either
+// the file can't be read or it contains an unknown plugin.
+func (pa *PluginAgent) Load(path string) error {
+	pa.mut.Lock()
+	defer pa.mut.Unlock()
+	b, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
+	}
+	np := map[string][]string{}
```

Comment:
**n**ew **p**lugin.

I think Erick was a fan of my names in this package too :)